### PR TITLE
📖 Assorted docs updates

### PR DESCRIPTION
**Pull request by moshloop**
/assign @ncdc @chuckha

### CODE REVIEW

File: docs/book/getting_started/interoperability.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+CAPBK will generate the certificates and keys used for bootstrapping a cluster if they don't exist.  CAPBK will also reuse existing certificates provided they follow this convention:
+
+* Each certificate must be stored in a single secret named one of:
+    * `<cluster name>-ca`
+    * `<cluster name>-etcd`
+    * `<cluster name>-proxy`
+    * `<cluster name>-sa`
+* Each secret must have the following label:
+  * `cluster.x-k8s.io/cluster-name=<cluster name>`
+* Secrets must be of type TLS with 2 values `tls.crt` and `tls.key`
```
Comment (ncdc) [inline docs/book/getting_started/interoperability.md]: Should we specify their encoding format (e.g. PEM)? Is it worth having a dummy example in here?

**Comment by moshloop** [comment]
@vincepri will do

File: docs/book/getting_started/interoperability.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
...RFFZSktvWklodmNOQVFFQkJRQURnZ0VQQURDQ0FRb0NnZ0VCQU1oZgogICAgV0lxdGUzMFRkNXlXM1hQWCs5SDV5S3I1N0NoNDJ2Y3o4cU9DdXFZazZMeWpBcG1hS3FyaTcwRTJ1ZVlYQ0U4RAogICAgMVAwaVRlOXd5L2M0VFpNcmFwUHBHNHo5UDF3bCtTdG5WcTVBZWV6aU9IY1BIcERyNVR4VGViUllnR3dmUFlLdwogICAgdm5kS0JseXYzQXhVR0NIU2hOV01hSUx0dko0RktKZ0NGNXVVUjE3SExsMnVyQ1dmNTBpNmEwK3pLcHZLejZYVwogICAgUmlGTWxZalhlUHFrKzB2QjdWcGxla05JTnFEbm9IRk1kZFBudXZjUW5NSTU1Q2NadHd5S09HMHp0OXBPYXdJYwogICAga3B5a0xSb20rWlBpQ3AzdG1BMnRBUHRsQmJTcmMvZmtVN1BUS2VacjduZ05RYnJocTFudlR3KzFYcmRwcG5qdQogICAgVTB6QVZ2N1JKOS9JSEJTdEFIY0NBd0VBQWFNbU1DUXdEZ1lEVlIwUEFRSC9CQVFEQWdLa01CSUdBMVVkRXdFQgogICAgL3dRSU1BWUJBZjhDQVFBd0RRWUpLb1pJaHZjTkFRRUxCUUFEZ2dFQkFKOUFUT01STC9hdnJGMWZ4N21xaDFFUwogICAgQ3JjcmZIQVFvWlhaTEhXV2lqVm56VGQvYmFKUTZhL0haejMvTm1TemxtK255cW1vWUNqODZmWENDUTdqLzdLbQogICAgMnk3YzhrVWwxSjFQMWtiNGpkWmhlQS9aa1ZWUHVpSHlRNU1CM3dVWEtDQTdudWtLem04blRKRUFWSXE0azEzdwogICAgL1MwYVZaM1JPVWpvSGxhcWl0L1NHdzZSVm9KQTVWb2p5Unh2VEIxdDFyL3JqajlFa0JLVkVHM3pQNGJacC8rcAogICAgdTdYWDluc0xRUWR5YVZSazU4N29vWXpDMTdOQys3OFJub1JoR2tXbUlZTmZWaUllSGVmNU9ac2UxZ0VPRXVOQQogICAgVW9WanlsRkJSbk9XL1pTVEN5bEhGU3M3Z0kvakV1cmVzbWtNdFYzTThxWG5LN2JaRG9rNzZpRGU1RDZlYUNzPQogICAgLS0tLS1FTkQgQ0VSVElGSUNBVEUtLS0tLQ==
```
[the remark this answers is not in the document]
Comment (moshloop) [reply docs/book/getting_started/interoperability.md]: This is  BASE64 encoded PEM

File: docs/book/getting_started/interoperability.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+Cluster API expects certificates and keys used for bootstrapping to follow the below convention. CAPBK generates new certificates using this convention if they do not already exist.
+
+* Each certificate must be stored PEM encoded in a single secret named one of:
+    * `<cluster name>-ca`
+    * `<cluster name>-etcd`
+    * `<cluster name>-proxy`
+    * `<cluster name>-sa`
+* Each secret must have the following label:
+  * `cluster.x-k8s.io/cluster-name=<cluster name>`
```
[the remark this answers is not in the document]
Comment (moshloop) [reply docs/book/getting_started/interoperability.md]: Would that not require us to list every secret and then filter (potentially triggering unnecessary audit events) - with the label we can query just the secrets for the specific cluster?

File: docs/book/getting_started/interoperability.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+### Certificate Management
+
+Cluster API expects certificates and keys used for bootstrapping to follow the below convention. CAPBK generates new certificates using this convention if they do not already exist.
+
+* Each certificate must be stored PEM encoded in a single secret named one of:
+    * `<cluster name>-ca`
+    * `<cluster name>-etcd`
+    * `<cluster name>-proxy`
+    * `<cluster name>-sa`
```
[the remark this answers is not in the document]
Comment (neolit123) [reply docs/book/getting_started/interoperability.md]: i'd go into more detail about the contents of these Secrets and their purpose.

[the remark this answers is not in the document]
Comment (detiber) [reply docs/book/getting_started/interoperability.md]: One thing that just crossed my mind, is that outside of `<cluster-name>-ca` and a generated `<cluster-name>-kubeconfig`, the other values listed here are particular to the kubeadm bootstrapper, maybe we should not define the interoperability of those values here, but rather on the kubeadm bootstrapper?

**Comment by ncdc** [comment]
@moshloop Hi! Will you have some time soon to pick this back up?

**Comment by vincepri** [comment]
If it's possible, it'd be great to move this page under the developer guide of the new book

**Comment by moshloop** [comment]
I will have this done before the F2F next week.

**Comment by vincepri** [comment]
@moshloop Do you have time this week to get this document to the gitbook?

**Comment by vincepri** [comment]
Thanks!

**Comment by moshloop** [comment]
ping @ncdc @vincepri @detiber

**Comment by vincepri** [comment]
@moshloop Seems there are some conflicts that need to be resolved

File: docs/book/SUMMARY.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Managing Kubernetes Infrastructure with the Cluster API
```
Comment (vincepri) [inline docs/book/SUMMARY.md]: This file seems extra

File: docs/book/src/architecture/controllers/cluster.md [hunk 6]
```diff
@@ -1 +1,3 @@
 # Cluster
+<!-- TODO -->
+![](../../images/cluster-admission-cluster-controller.svg)
```
Comment (vincepri) [inline docs/book/src/architecture/controllers/cluster.md]: The image linked seems to be missing?

**Comment by vincepri** [comment]
@moshloop I'm not sure what happened to the docs, but it seems the changes in this PR aren't based on the latest version of the book, which is published here https://cluster-api.sigs.k8s.io. 

The scope of this PR has grown too much imo, and I want to ask if you have time to either split the commits or preferably to open different PRs so it's easier to review.

/hold

File: docs/book/src/user/installing.md [hunk 7]
```diff
@@ -0,0 +1,119 @@
+# Installation
```
Comment (vincepri) [inline docs/book/src/user/installing.md]: This is the quick start and shouldn't be renamed. There are multiple links to it in the wild

File: docs/book/src/SUMMARY.md [hunk 8]
```diff
@@ -2,13 +2,16 @@
 
 [Introduction](./introduction.md)
 
+- [Quick Start](./user/quick-start.md)
```
Comment (vincepri) [inline docs/book/src/SUMMARY.md]:
```suggestion
[Quick Start](./user/quick-start.md)
```

**Comment by moshloop** [comment]
@vincepri @randomvariable @ncdc 

Deployed to https://book-789v67wdn.now.sh/ for easy review

**Pull request merged**